### PR TITLE
fix: strategy migration only through Vault

### PR DESCRIPTION
**Pull request by steffenix**

### CODE REVIEW

File: contracts/BaseStrategy.sol [hunk 1]
```diff
@@ -757,13 +757,16 @@ abstract contract BaseStrategy {
      * @notice
      *  Transfers all `want` from this Strategy to `_newStrategy`.
      *
-     *  This may only be called by governance or the Vault.
+     *  This may only be called by the Vault.
      * @dev
-     *  The new Strategy's Vault must be the same as this Strategy's Vault.
+     * The new Strategy's Vault must be the same as this Strategy's Vault.
+     * the migration process should be carefully performed to make sure all
```
Comment (fubuloubu) [inline contracts/BaseStrategy.sol:763]:
```suggestion
     *  The migration process should be carefully performed to make sure all
```

File: tests/functional/vault/test_config.py [hunk 2]
```diff
@@ -80,7 +80,16 @@ def test_vault_reinitialization(guardian, gov, rewards, token, Vault):
     ],
 )
 def test_vault_setParams(
-    chain, gov, guardian, management, vault, rando, getter, setter, val, guard_allowed,
+    chain,
```
Comment (fubuloubu) [inline tests/functional/vault/test_config.py:83]: Seems like some black updated code in this PR

**Review by fubuloubu — APPROVED** [review]

File: tests/functional/vault/test_config.py (hunk 2 above)
Comment (steffenix) [reply tests/functional/vault/test_config.py:83]: This will be rebased on top of https://github.com/yearn/yearn-vaults/pull/310

**Comment by fubuloubu** [comment]
Fix last commit message to be consistent with commitlint and we can merge this

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**